docs/readmes/README.role-assignment.md

```diff
@@ -0,0 +1,47 @@
+# Role Assignment
+
+Without the proper role assignment, your Azure cluster will not have the correct permission to perform Azure-related operations such as creating and removing identity assignments.
+
+## Introduction
+
+The [MIC](../../README.md#managed-identity-controller) component in AAD Pod Identity needs to authenticate with Azure to create and remove identity assignment from the underlying virtual machines (VM) or virtual machine scale sets (VMSS). Currently, MIC will use one of the following two ways to autheticate with Azure:
+
+1. [Service principal](https://docs.microsoft.com/en-us/azure/aks/kubernetes-service-principal) through `/etc/kubernetes/azure.json` available in every node, or credential defined by environment variables;
+2. [Managed identity](https://docs.microsoft.com/en-us/azure/aks/use-managed-identity) (it can be using a system-assigned identity or user-assigned identity)
```

Comment:
`/etc/kubernetes/azure.json` provides the details about what to use for auth 

1. `useManagedIdentityExtension: true` and `userAssignedIdentityID:""` - system-assigned managed identity cluster
2. `useManagedIdentityExtension: true` and `userAssignedIdentityID:"xxx"` - user-assigned managed identity cluster
3. Default is fallback to using SP credentials

We should document, based on what the cluster uses for auth, MIC will use the same